design-reference/architecture/aad/easyauth.md

```diff
@@ -0,0 +1,35 @@
+# Purpose/Description
+
+Cobalts customers expect the ability to configure Authentication for their applications. We've chosen to provide this ability by extending both the Service Principal module and the App Service module. This will enable a customer to configure "Easy Auth" on their App Service using an application registration in Azure Active Directory.
+Azure App Service provides built-in authentication and authorization support, so you can sign in users and access data by writing minimal or no code in your web app, RESTful API, and mobile back end, and also Azure Functions. This article describes how App Service helps simplify authentication and authorization for your app.
+
```

Comment:
What's the guidance for REST clients using a system account / service principal?

Reply:
To piggy back here -- It would be great to have a discussion or data flow diagram that talks through how the check for "is this client authenticated" done in detail

Reply:
I feel like these are implementation details that pertain to how the code will work... not the resources needed to enable that code. What I'm hearing is this isn't clear.

Reply:
@iphilpot - my 2 cents... take it with a grain of salt. From the perspective of somebody who isn't familiar with the technology being suggested here what is not clear is how the resources being proposed will enable a client to authenticate. Something as lightweight as "the client needs to generate a token using XYZ and include that as a HTTP header" would help me understand why the proposed design meets the customer's requirements. I don't think we need pseudo code as I agree that would definitely venture into implementation details that are outside the scope.